[FEAT/#220] 커스텀 토스트 구현

boiledeggg opened the pull request:
## Related issue 🛠
- closed #220 

## Work Description ✏️
- `NapzakToast` 구현
- `SnackBarController` 클래스 제거
-  물품상세, 신고 화면에 NapzakToast 적용
- MainScreen에 NapzakToast 적용

## Screenshot 📸
| 홈화면 뒤로가기 | 상세화면 좋아요 | 상세화면 상품상태변경 |
|:-------------:|:-------------:|:-------------:|
| <img src="https://github.com/user-attachments/assets/84aa98d3-3601-49aa-868a-ab405d2909da" width="300"/> | <img src="https://github.com/user-attachments/assets/9a09cfab-d70e-4472-9266-15669f2e8449" width="300"/> | <video src="https://github.com/user-attachments/assets/f56456bf-a6e0-456f-a427-a4b84ada4196" width="300"></video> |

## To Reviewers 📢
- 커스텀 토스트 객체 NapzakToast를 구현했습니다! 
- 자세한 사용법에 대해선 노션 페이지에 정리해둘테니 참고하셔서 현재 사용 중인 스낵바를 다 NapzakToast로 대체합시닷~!!

# [NapzakToast 사용법 보러가기](https://www.notion.so/NapzakToast-204f54d645db80cd92e1e27e6edd5623?source=copy_link)

boiledeggg committed:
[move/#220] 스낵바 컴포넌트 위치 수정

boiledeggg committed:
[feat/#220] 커스텀 토스트 클래스 구현

boiledeggg committed:
[chore/#220] TradeStatusType "판매중" 라벨 추가

boiledeggg committed:
[feat/#220] NapzakToast 전역변수 추가

boiledeggg committed:
[feat/#220] 상세페이지에 Toast 적용

- SideEffect 객체를 추가 구현
- SnackBar를 제거하고 Toast로 대체

boiledeggg committed:
[feat/#220] 상세페이지 바텀시트에서 SnackBar 제거

boiledeggg committed:
[rename/#220] 상세페이지 문자열 리소스 이름 수정

boiledeggg committed:
[refactor/#220] NapzakToast 리팩터링

- 파라미터 수정
- Napzak 폰트 스타일 클래스 추가

boiledeggg committed:
[mod/#220] 신고화면 스낵바를 토스트로 대체

boiledeggg committed:
[rename/#220] Toast 폰트 파라미터 이름 수정

boiledeggg committed:
[del/#220] SnackBarController를 제거하고 NapzakToast로 대체

boiledeggg committed:
[del/#220] commonToast 오프셋 파라미터 추가

boiledeggg committed:
[docs/#220] NapzakToast 주석 추가

chrin05 reviewed (approved):
고생하셨습니다! 짱짱맨~
사용법 참고해서 제 부분에도 적용하겠습니다

boiledeggg committed:
[refactor/#220] NapzakToast 코드 리팩터링

- NapzakToast 호출 함수를 하나로 제한
- ToastType을 구현하여 토스트메시지 디자인 구분
- Screen 및 뷰모델 파일 수정

chrin05 reviewed (approved):
변경된 부분 확인했습니다!! 깔끔해졌네요 수고하셨습니다!!

jm991014 reviewed (approved):
크 역시 완벽하네요 납작 토스트 코드 좋아요~
